### ISSUE TITLE
screen freezes after calling navigate function with 'key'

### ISSUE THREAD

**Issue by ryanix** [react-navigation/react-navigation#3477]
### Current Behavior

- What code are you running and what is happening?
  Inside StackNavigator I had three screens registered like this
```javascript
const Stack = StackNavigator({
  Home: {
    screen: HomeScreen
  },
  Home1: {
    screen: HomeScreen1
  },
  Home2: {
    screen: HomeScreen2
  }
},{
  headerMode: 'none',
});
```
on each screen I had a button which will navigate to next screen when pressed, on the last screen I had a button navigate to the first screen
When I navigate back to 'Home' screen on 'Home2' screen with 
```javascript
navigation.navigate({routeName, key}) 
```
I used a global variable routeNameList to record the route key of each stack,

routeNameList looks like this { routeName: key }

it successfully navigate back, but the screen freeze,  button no longer responds to click

- Include a screenshot if it makes sense.

### Expected Behavior

- What do you expect should be happening?
Screen should not freeze
- Include a screenshot if it makes sense.

### How to reproduce

full code is attached:
```javascript
import React, { Component } from 'react';
import {  AppRegistry,View,Button} from 'react-native';
import {StackNavigator} from 'react-navigation';


var routeNameList = {}
function navigateTo(nav,route) {
  let key = routeNameList[route]
  if (key) {
    nav.navigate({routeName: route, key:key})
  } else {
    nav.navigate(route)
  }
}

function register(props) {
  let route = props.navigation.state;
  let {routeName, key} = route;
  routeNameList[routeName] = key
}


class HomeScreen extends React.Component {
  render() {
    register(this.props)
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Button title='Jump To Home1'  onPress={()=> navigateTo(this.props.navigation,'Home1')}/>
      </View>
    );
  }
}

class HomeScreen1 extends React.Component {
  render() {
    register(this.props)
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Button title='Jump To Home2'  onPress={()=> navigateTo(this.props.navigation,'Home2')}/>
      </View>
    );
  }
}

class HomeScreen2 extends React.Component {
  render() {
    register(this.props)
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Button title='Jump back To Home'  onPress={()=> navigateTo(this.props.navigation,'Home')}/>
      </View>
    );
  }
}


const Stack = StackNavigator({
  Home: {
    screen: HomeScreen
  },
  Home1: {
    screen: HomeScreen1
  },
  Home2: {
    screen: HomeScreen2
  }
},{
  headerMode: 'none',
});

AppRegistry.registerComponent("HappsNow",()=>Stack);
```

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 1.0.2
| react-native     | 0.52.2
| node             | v9.4.0
| npm or yarn      | 5.6.0

**Comment by brentvatne** [comment]
example on Snack: https://snack.expo.io/HJrXLu58M

**Comment by brentvatne** [comment]
fixed and publish in 1.0.3

**Comment by brentvatne** [comment]
also, here is the same example as above on 1.0.3 so you can see it working: https://snack.expo.io/SJGD_t58M
